### PR TITLE
docs: Update HACKING guide with recommended git config

### PR DESCRIPTION
**Pull request by daandemeyer**

### CODE REVIEW

**Comment by poettering** [comment]
So we have tools/add-git-hook.sh and hooked that into meson. Shouldn't the same code also just toggle these options?

i.e. if we already reach over from meson into git settings, we might as well add too more?

File: docs/HACKING.md [hunk 1]
```diff
@@ -14,6 +14,14 @@ feature, please feel invited to fix it, and submit your work as a
...
 patches. Also have a look at our [Contribution Guidelines](CONTRIBUTING).
 
+To start, run the following commands in the systemd git repository to set up
+git correctly:
+
+```shell
+$ git config submodule.recurse true
+$ git config fetch.recurseSubmodules on-demand
+```
+
```
Comment (poettering) [inline docs/HACKING.md:24]: this can be dropped now, no?

**Comment by poettering** [comment]
will leave the review of this to @keszybz, who last worked on hooking this in in 003c8879679e21ea4b9f76770cc9a23d24759cb0

File: docs/HACKING.md (hunk 1 above)
Comment (bluca) [reply docs/HACKING.md:24]: I think it's better to document explicitly, not everyone runs scripts from repos they clone

Comment (poettering) [reply docs/HACKING.md:24]: meson hooks this in automatically... so if you run meson you are already set.

At the very least this is something to mention: that the meson build stuff will enable this for you

Comment (daandemeyer) [reply docs/HACKING.md:24]: @poettering Yes but it's perfectly possible to hack on systemd these days without ever running `meson` on the host at all. And changes to sources from `meson` running in mkosi are thrown away at the end of the build. So it's not guaranteed that these settings will always be applied so documenting them as well seems like a good idea.

Comment (keszybz) [reply docs/HACKING.md:24]: But then what @poettering said, it should be mentioned that running those commands is not necessary if meson is used. Maybe, "If you're not using meson at all, …".

**Pull request merged**